enforcer/file/common/src/test/groovy/com/optum/sourcehawk/enforcer/file/json/JsonValueEqualsSpec.groovy

```diff
@@ -0,0 +1,379 @@
+        InputStream fileInputStream = IoUtil.getResourceAsStream('/bicycle.json')
+        StringWriter stringWriter = new StringWriter()
+
+        when:
+        ResolverResult result = jsonPathEquals.resolve(fileInputStream, stringWriter)
+
+        then:
+        result
+        !result.updatesApplied
+        result.fixCount == 0
+        !result.error
+        result.errorCount == 0
+        !result.messages
+
+        and:
+        !stringWriter.toString()
+
+        where:
+        pointerExpression | expectedValue
+        '/make'           | 'Raleigh'
+        '/size/value'     | 60
+        '/components/0'   | 'handlebars'
+    }
+
+    @Unroll
+    def "resolve - updates applied (pointer expression found)"() {
+        given:
+        JsonValueEquals jsonPathEquals = JsonValueEquals.equals(pointerExpression, expectedValue)
+        InputStream fileInputStream = IoUtil.getResourceAsStream('/bicycle.json')
+        StringWriter stringWriter = new StringWriter()
+
+        when:
+        ResolverResult result = jsonPathEquals.resolve(fileInputStream, stringWriter)
+
+        then:
+        result
+        result.updatesApplied
+        result.fixCount == 1
+        !result.error
+        result.errorCount == 0
+        result.messages
+        result.messages.size() == 1
+        result.messages[0] == "Pointer expression [$pointerExpression] has been updated with value [$expectedValue]"
+
+        and:
+        stringWriter.toString()
+
+        where:
+        pointerExpression | expectedValue
+        '/make'           | 'Cinelli'
+        '/size/value'     | 61
+        '/components/0'   | 'stem'
+    }
+
+    @Unroll
+    def "resolve - updates applied (pointer expression not found)"() {
+        given:
+        JsonValueEquals jsonPathEquals = JsonValueEquals.equals(pointerExpression, expectedValue)
+        InputStream fileInputStream = IoUtil.getResourceAsStream('/bicycle.json')
+        StringWriter stringWriter = new StringWriter()
+
+        when:
+        ResolverResult result = jsonPathEquals.resolve(fileInputStream, stringWriter)
+
+        then:
+        result
+        result.updatesApplied
+        result.fixCount == 1
+        !result.error
+        result.errorCount == 0
+        result.messages
+        result.messages.size() == 1
+        result.messages[0] == "Pointer expression [$pointerExpression] which was missing, has been set with value [$expectedValue]"
+
+        when:
+        Map jsonObject = new ObjectMapper().readValue(stringWriter.toString(), Map)
+
+        then:
+        expectedJsonAssertion.apply(jsonObject)
+
+        where:
+        pointerExpression | expectedValue | expectedJsonAssertion
+        '/rating'         | 98            | { json -> json.rating == 98 } as Function<Map, Boolean>
+//        '/notes/0'        | 'Note 1'      | { json -> json.notes[0] == "Note 1" } as Function<Map, Boolean> (array addition not yet supported)
```

Comment:
curious what if the ACTUAL node key is a number like thing ie json.notes.0 = "Note 1" does this still work?

```json
{
"notes": {
"0": "Note1"
},
{"1": Note 2"
}
}
```
etc

Reply:
same for above items using the /0 or expected list indexor

Reply:
I'll add a test for it.

Reply:
Tested, and it works.  Pushed an updated test.  It works because there is a concept of an "object" node and an "array" node.